src/sinks/vector/sink.rs

```diff
@@ -43,11 +44,31 @@ where
             .batched(self.batch_settings.into_reducer_config(
                 |data: &EventData| data.wrapper.encoded_len(),
                 |req: &mut VectorRequest, item: EventData| {
-                    req.events_byte_size += item.byte_size;
                     req.finalizers.merge(item.finalizers);
                     req.events.push(item.wrapper);
+                    req.events_byte_size += item.byte_size;
                 },
             ))
+            .map(|mut v_request| {
+                let builder = RequestMetadataBuilder::new(
+                    v_request.events.len(),
+                    v_request.events_byte_size,
+                    v_request.events_byte_size, // this is fine as it isn't being used
+                );
+
+                let encoded_events = proto_vector::PushEventsRequest {
+                    events: v_request.events.clone(),
+                };
```

Comment:
Is this cloning the entire request data per-request? That seems like a performance pessimization we'd like to avoid.

Reply:
Yeah I don't like the clone here either... the proto definition doesn't accept a `&` though :thinking: so I'm not exactly sure how to not clone since we need to return the `v_request` itself at the end of the map.

Reply:
Per our offline discussion I introduced a new temporary struct for the event collection only and this removed the need for the expensive clone- good recommendation!